DKTDDSwiftUI/Greetings.swift

```diff
@@ -7,19 +7,19 @@
 
 import Foundation
 
-class Greetings {
+struct Greetings {
     // MARK: Properties
-    var result = ""
-    var nameWithCommas = [String]()
-    var lastName = ""
-    var namesString = ""
+    private var result = ""
+    private var nameWithCommas = [String]()
+    private var lastName = ""
+    private var namesString = ""
     
     // MARK: Methods
     /// Greet takes a string  with names.
     /// - Parameters:
     ///   - name: String name
     /// - Returns: return a string "Hello, name" if one name  || "Hello, my friend" if empty string or nil || "HELLO, NAME" if uppercase name
-    func greet(name: String? = nil) -> String {
+    mutating func greet(name: String? = nil) -> String {
```

Comment:
Mutating funcs will still leave us with state changes that may not be necessary. Normally mutating funcs to me means we should have a class. Is there a way to change your code to remove the need for state entirely?

Reply:
Yes there's a much better way that will improve my code so much more and will get rid-off inout.  This solution don't require @State or @Binding properties either.